pkg/webui/components/button/copy-button/index.js

```diff
@@ -0,0 +1,112 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import bind from 'autobind-decorator'
+import clipboard from 'clipboard'
+
+import Message from '@ttn-lw/lib/components/message'
+
+import PropTypes from '@ttn-lw/lib/prop-types'
+import sharedMessages from '@ttn-lw/lib/shared-messages'
+
+import Button from '..'
+
+/**
+ * ModalButton is a button which needs a modal confirmation to complete the
+ * action. It can be used as an easy way to get the users explicit confirmation
+ * before doing an action, e.g. Deleting a resource.
+ */
+class CopyButton extends React.Component {
+  static propTypes = {
+    clipboardText: PropTypes.string.isRequired,
+    copiedMessage: PropTypes.message,
+    noCopyPopup: PropTypes.bool,
+    onClick: PropTypes.func,
+  }
+
+  static defaultProps = {
+    copiedMessage: sharedMessages.copied,
+    noCopyPopup: false,
+    onClick: () => {},
+  }
+
+  constructor(props) {
+    super(props)
+
+    this._timer = null
+
+    this.state = {
+      copied: false,
+    }
+
+    this.copyElem = React.createRef()
+  }
```

Comment:
```suggestion
   timer = null
   copyElem = React.createRef()
   state = {
      copied: false,
    }
```

The constructor is not needed here. Note though that this has to be refactored anyway into a functional component.